influxdb3_write/src/lib.rs

```diff
@@ -343,3 +344,27 @@ pub struct ParquetFile {
     pub min_time: i64,
     pub max_time: i64,
 }
+
+/// The summary data for a persisted parquet file in a segment.
+#[derive(Clone, Copy, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum Precision {
+    Auto,
+    Second,
+    Millisecond,
+    Microsecond,
+    Nanosecond,
+}
+
+pub(crate) fn guess_precision(timestamp: i64) -> Precision {
```

Comment:
This function looks deserving of a brief comment to explain the logic.

Some points worth mentioning:
* `timestamp` is a unix time of arbitrary precision
* The comment about it failing in June 2128, lol! (which you comment on elsewhere, but I think belongs here)
* It doesn't work for negative timestamps - ultimately, we could have a requirement that writes with negative timestamps specify precision

Reply:
I solved it by using the absolute value of the number so that we can use it for both negatives and positive numbers. I've also updated the comments for this so that it's more clear what's going on.